### PR TITLE
Create index.d.ts for use in TS

### PR DESCRIPTION
**Pull request by daniyel**

### CODE REVIEW

**Comment by daniyel** [comment]
Update index.d.ts

**Comment by RecuencoJones** [comment]
Hi @daniyel, could you please link the definitions in the `package.json` with the `typings` entry?

File: index.d.ts [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+        auth?: Auth
+
+        /** Agent for the request */
+        agent?: http.Agent|https.Agent
+
+        /** Context for substitution */
+        context?: { [key: string]: any }
+    }
+
+    interface LoadCallback {
```
Comment (RecuencoJones) [inline index.d.ts:92]: Maybe we could use the following instead:

```
interface LoadCallback {
  (error: Error, config?: Config): void
}
```

File: index.d.ts [hunk 2]
```diff
@@ -0,0 +1,103 @@
+/// <reference types="node" />
+
+declare module 'cloud-config-client' {
+
+    import http = require('http');
```
Comment (RecuencoJones) [inline index.d.ts:5]: Let's use `;` everywhere or not use them at all

File: index.d.ts [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+
+    interface ConfigFile {
+        /** file name */
+        name: string
+
+        /** configuration properties */
+        source: ConfigSource
+    }
+
+    interface ConfigSource {}
```
Comment (RecuencoJones) [inline index.d.ts]: Let's use a type instead of an interface here 

```
type ConfigSource {
  [key: string]: any
}
```

File: index.d.ts [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+        label: string
+
+        /** commit hash of properties */
+        version: string
+
+        /** properties included for application, sorted by more priority */
+        propertySources: Array<ConfigFile>
+    }
+
+    interface Spring {
```
Comment (RecuencoJones) [inline index.d.ts]: I don't think this interface is actually necessary. I guess it comes from the [Spring module definition](https://github.com/victorherraiz/cloud-config-client/blob/master/lib/spring.js#L4) which I think we can skip here

**Review by RecuencoJones — COMMENTED** [review]
Thanks @daniyel for creating the type definitions! I added some suggestions but so far it looks great 😄

File: index.d.ts (hunk 1 above)
Comment (daniyel) [reply index.d.ts:92]: Yes sure, you can adjust as you think it is the best.

File: index.d.ts (hunk 2 above)
Comment (daniyel) [reply index.d.ts:5]: Yes sure, you can adjust as you think it is the best.

File: index.d.ts (hunk 3 above)
Comment (daniyel) [reply index.d.ts]: Yes sure, you can adjust as you think it is the best.

File: index.d.ts (hunk 1 above)
Comment (daniyel) [reply index.d.ts:92]: I have update this one.

File: index.d.ts (hunk 2 above)
Comment (daniyel) [reply index.d.ts:5]: This one I am not sure where do you mean `;` are still missing. I kind of referenced myself to `mongoose` types.

**Comment by daniyel** [comment]
I have added `types` in `package.json`

**Comment by daniyel** [comment]
Hi. Any more wishes, request, improvements you have?

**Comment by oktapodia** [comment]
Great work, any news on it? @victorherraiz

**Review by isaacdomini — APPROVED** [review]

**Pull request merged**